src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -80,19 +81,55 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 
 	r.LastObservedState = &sts
 
-	// Ensure StatefulSet #replicas equals cluster requirement.
-	if sts.Spec.Replicas != nil && r.pandaCluster.Spec.Replicas != nil && *sts.Spec.Replicas != *r.pandaCluster.Spec.Replicas {
-		r.logger.Info(fmt.Sprintf("StatefulSet %s has replicas set to %d but need %d. Going to update", r.Key().Name, *sts.Spec.Replicas, *r.pandaCluster.Spec.Replicas))
-
-		sts.Spec.Replicas = r.pandaCluster.Spec.Replicas
+	updated := Update(&sts, r.pandaCluster, r.logger)
+	if updated {
 		if err := r.Update(ctx, &sts); err != nil {
-			return fmt.Errorf("failed to update StatefulSet: %w", err)
+			return fmt.Errorf("failed to Update StatefulSet: %w", err)
 		}
 	}
 
 	return nil
 }
 
+// Update ensures StatefulSet #replicas and resources equals cluster requirements.
+func Update(
+	sts *appsv1.StatefulSet,
+	pandaCluster *redpandav1alpha1.Cluster,
+	logger logr.Logger,
+) (updated bool) {
+	return updateReplicasIfNeeded(sts, pandaCluster, logger) || updateResourcesIfNeeded(sts, pandaCluster, logger)
+}
+
+func updateResourcesIfNeeded(
+	sts *appsv1.StatefulSet,
+	pandaCluster *redpandav1alpha1.Cluster,
+	logger logr.Logger,
+) (updated bool) {
+	if !reflect.DeepEqual(sts.Spec.Template.Spec.Containers[0].Resources, pandaCluster.Spec.Resources) {
```

Comment:
yeah I can safeguard against this but then ... this should never happen :D since we're owning that object. But if you're asking whether someone CAN submit STS that will have no containers, I am not sure 🤔 

EDIT: okay, it cannot be empty

```error: error validating "ss.yaml": error validating data: ValidationError(StatefulSet.spec.template.spec): missing required field "containers" in io.k8s.api.core.v1.PodSpec; if you choose to ignore these errors, turn validation off with --validate=false```